### ISSUE TITLE
学习PaddleHub使用预训练模型和Finetune的工具 Loss值很大

### ISSUE THREAD

**Issue by qingchunnianshao** [PaddlePaddle/Paddle#54091]
### 问题描述 Please describe your issue

在学习PaddleHub使用预训练模型和Finetune的工具 使用自己的数据进行fine-tune时，loss 结果不正确，loss一直很大还存在负值情况

版本、环境信息
1）PaddleHub2.1，PaddlePaddle2.4.2
2）系统环境：使用的是window系统，cuda10.2，cudnn7.6与cudnn8.0都有尝试问题一样
以下为我的代码块：
```

import os
import cv2
import paddlehub as hub
import matplotlib.pyplot as plt
import matplotlib.image as mpimg

import paddle
import paddle.vision.transforms as T
from paddlehub.finetune.trainer import Trainer

from PIL import Image

os.environ['CUDA_VISIBLE_DEVICES']='0'

class DemoDataset(paddle.io.Dataset):
    def __init__(self, transforms, num_classes=4, mode='train'):
        # 数据集存放位置
        self.dataset_dir = "./peach/peach-classification"  # dataset_dir为数据集实际路径，需要填写全路径
        self.transforms = transforms
        self.num_classes = num_classes
        self.mode = mode

        if self.mode == 'train':
            self.file = 'train_list.txt'
        elif self.mode == 'test':
            self.file = 'test_list.txt'
        else:
            self.file = 'validate_list.txt'

        self.file = os.path.join(self.dataset_dir, self.file)
        with open(self.file, 'r') as file:
            self.data = file.read().split('\n')[:-1]

    def __getitem__(self, idx):
        img_path, grt = self.data[idx].split(' ')
        img_path = os.path.join(self.dataset_dir, img_path)
        #print(img_path)
        img = cv2.imread(img_path)
        img = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)
        img = img.astype('float32')
        img = img/255.0
        im = self.transforms(img)
        return im, int(grt)

    def __len__(self):
        return len(self.data)

if __name__ == '__main__':
    print('fine-tune')
    transforms = T.Compose(
        [T.Resize((256, 256)),
         T.CenterCrop(224),
         T.Transpose((2,0,1)),
         T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])],
        )

    peach_train = DemoDataset(transforms)
    img,t = peach_train[0]
    peach_validate = DemoDataset(transforms, mode='val')
    model = hub.Module(name='resnet50_vd_imagenet_ssld', label_list=["R0", "B1", "M2", "S3"])

    optimizer = paddle.optimizer.Adam(learning_rate=0.001, parameters=model.parameters())
    trainer = Trainer(model, optimizer, checkpoint_dir='img_classification_ckpt', use_gpu=True)
    trainer.train(peach_train, epochs=10, batch_size=16, eval_dataset=peach_validate, save_interval=1)

    result = model.predict(['./peach/peach-classification/test/M2/0.png'])
    print(result)
```
10个epoch结束依然发现loss在很大范围内波动
训练的输出为：
C:\ProgramData\Anaconda3\envs\pythonProject\lib\site-packages_distutils_hack_init_.py:33: UserWarning: Setuptools is replacing distutils.
warnings.warn("Setuptools is replacing distutils.")
fine-tune
W0524 17:55:10.098229 19612 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.6, Driver API Version: 11.7, Runtime API Version: 10.2
W0524 17:55:10.105249 19612 gpu_resources.cc:91] device: 0, cuDNN Version: 7.6.
load pretrained parameters success
[2023-05-24 17:55:13,402] [ WARNING] - PaddleHub model checkpoint not found, start from scratch...
[2023-05-24 17:55:18,730] [ TRAIN] - Epoch=1/10, Step=10/375 loss=-128960672122.5853 acc=0.0000 lr=0.001000 step/sec=2.39 | ETA 00:26:11
[2023-05-24 17:55:20,207] [ TRAIN] - Epoch=1/10, Step=20/375 loss=-12402729.9590 acc=0.0000 lr=0.001000 step/sec=6.77 | ETA 00:17:42
[2023-05-24 17:55:21,697] [ TRAIN] - Epoch=1/10, Step=30/375 loss=235926904.1966 acc=0.0000 lr=0.001000 step/sec=6.71 | ETA 00:14:54
[2023-05-24 17:55:23,129] [ TRAIN] - Epoch=1/10, Step=40/375 loss=64.1373 acc=0.0000 lr=0.001000 step/sec=6.98 | ETA 00:13:25
[2023-05-24 17:55:24,606] [ TRAIN] - Epoch=1/10, Step=50/375 loss=-181.9779 acc=0.0000 lr=0.001000 step/sec=6.77 | ETA 00:12:35
[2023-05-24 17:55:26,148] [ TRAIN] - Epoch=1/10, Step=60/375 loss=33682076467.9539 acc=0.0000 lr=0.001000 step/sec=6.48 | ETA 00:12:05

期待能够指出我的问题所在

**Comment by wangna11BD** [comment]
可以输出中间结果看看是否符合预期，定位一下问题。loss过大的原因比较多，有可能是网络结构设计不合适，数据集处理又问题，loss不合适等等，需要您自己去调整。

**Comment by qingchunnianshao** [comment]
> 可以输出中间结果看看是否符合预期，定位一下问题。loss过大的原因比较多，有可能是网络结构设计不合适，数据集处理又问题，loss不合适等等，需要您自己去调整。

是这样的，我在AIStudio上直接训练是能得到预期的效果的，只是在本地配置环境，使用相同的数据结果就不正确，不知道问题出在哪里，使用的数据集就是课程里面《预训练模型应用工具paddleHub》那节课中的数据
有一点不同的地方就是 修改了下面这个函数，上面是我修改的，下面是原来的，transforms也进行了一点修改，直接使用课程中的transforms 会提示报错提示没有to_rgb属性

看看能否帮忙定位问题大致方向


```

transforms = T.Compose(
        [
        T.Resize((256, 256)),
         T.CenterCrop(224),
         T.Transpose((2,0,1)),
         T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
        ]
        )
transforms = T.Compose(
        [T.Resize((256, 256)),
         T.CenterCrop(224),
         T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])],
        to_rgb=True)

def __getitem__(self, idx):
        img_path, grt = self.data[idx].split(' ')
        img_path = os.path.join(self.dataset_dir, img_path)
        #print(img_path)
        img = cv2.imread(img_path)
        img = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)
        img = img.astype('float32')
        img = img/255.0
        im = self.transforms(img)
        return im, int(grt)

def __getitem__(self, idx):
        img_path, grt = self.data[idx].split(' ')
        img_path = os.path.join(self.dataset_dir, img_path)
        im = self.transforms(img_path)
        return im, int(grt)
```
[2023-05-24 16:36:07,483] [ WARNING] - PaddleHub model checkpoint not found, start from scratch...
[2023-05-24 16:36:12,230] [   TRAIN] - Epoch=1/10, Step=10/375 loss=1.0124 acc=0.6125 lr=0.001000 step/sec=2.11 | ETA 00:29:38
[2023-05-24 16:36:13,686] [   TRAIN] - Epoch=1/10, Step=20/375 loss=0.5179 acc=0.8062 lr=0.001000 step/sec=6.87 | ETA 00:19:22
[2023-05-24 16:36:15,156] [   TRAIN] - Epoch=1/10, Step=30/375 loss=0.4418 acc=0.8438 lr=0.001000 step/sec=6.80 | ETA 00:15:58
[2023-05-24 16:36:16,617] [   TRAIN] - Epoch=1/10, Step=40/375 loss=0.1953 acc=0.9438 lr=0.001000 step/sec=6.84 | ETA 00:14:15
[2023-05-24 16:36:17,854] [   TRAIN] - Epoch=1/10, Step=50/375 loss=0.1276 acc=0.9750 lr=0.001000 step/sec=8.08 | ETA 00:12:57
[2023-05-24 16:36:18,883] [   TRAIN] - Epoch=1/10, Step=60/375 loss=0.1137 acc=0.9625 lr=0.001000 step/sec=9.72 | ETA 00:11:52

**Comment by qingchunnianshao** [comment]
cuda cudnn 显卡驱动版本不匹配的问题
